go.work

```diff
@@ -0,0 +1,7 @@
+go 1.18
```

Comment:
Committing go.work is an opinionated change that prevents other users from using go workspaces. E.g.: I use go workspaces to link Terraform bridge, Pulumi pkg/sdk, and a provider.

This change would result in a user's IDE ignoring their workspaces, as the nearest go.work for determining workspace behavior.

```
❯ go env GOWORK
 /home/friel/c/go.work
 
 ❯ cat /home/friel/c/go.work
 go 1.18
 
 use (
         ./github.com/pulumi/pulumi-terraform-bridge
         ./github.com/pulumi/pulumi-yaml
         ./github.com/pulumi/pulumi/pkg
         ./github.com/pulumi/pulumi/sdk
         ./github.com/pulumi/pulumi/tests
         ./github.com/pulumi/pulumi-docker/provider
# ...
 )
```

Reply:
Understood! Yeah, I learned about `go.work` just yesterday to try and solve this problem. There's an [open issue](https://github.com/golangci/golangci-lint/issues/2654) in golangci-lint which is related; running `golangci-lint run ./pkg ./tests ./sdk` only works if you have a `go.work` file, and that file must contain those three modules.

I wasn't sure if `go.work` was opinionated or not.

This PR will be blocked until `golangci-lint run ./pkg ./tests ./sdk` works again, without `go.work`.

Reply:
@AaronFriel I added a hack that creates a go.work file if one doesn't exist, but I don't think this solves your issue.